src/epoll.rs

```diff
@@ -42,7 +63,17 @@ impl EpollWrapper {
             let _ = self
                 .epoll
                 .ctl(ControlOperation::Delete, fd, EpollEvent::default());
-            self.fd_dispatch.remove(&fd);
+            self.remove_fd(fd);
+        }
+    }
+
+    // Remove the file descriptor from the dispatch table and pending fd set.
+    pub(crate) fn remove_fd(&mut self, fd: RawFd) {
+        self.fd_dispatch.remove(&fd);
+        for event in self.ready_events.iter_mut() {
```

Comment:
I added a commit on top of your commits that moves this code to a new structure (ReadyList). We can later use this abstraction to fix the TODO you added related to the EpollWrapper iterator.

Lemme me know if you'd like to introduce this to your PR: https://github.com/andreeaflorescu/event-manager/commit/d81e0f122d6f51214dc33cc533aaf7405648c432

Otherwise, I can also add this in a subsequent PR.

Reply:
How about combining it with the coming patch to solve the TODO?